suite/tests/client-interface/stolen-reg.dll.c

```diff
@@ -67,6 +67,59 @@ do_flush(app_pc next_pc)
     DR_ASSERT(false);
 }
 
+// Storing the original stolen reg before mconext set.
+static ptr_int_t orig_value = 0;
+
+// Random value to detect after mcontext set.
+static ptr_int_t test_value = 7;
+
+static void
+load_stolen_reg_to_mcontext()
+{
+    fprintf(stderr, "load_stolen_reg_to_mcontext entered\n");
+    void *drcontext = dr_get_current_drcontext();
+
+    fprintf(stderr, "test value = " IF_ARM_ELSE("%d", "%ld") "\n", test_value);
+
+    dr_mcontext_t mc;
+    mc.size = sizeof(mc);
+    mc.flags = DR_MC_ALL;
+
+    fprintf(stderr, "fetching TLS\n");
+
+    dr_get_mcontext(drcontext, &mc);
+
+    fprintf(stderr, "mc->stolen_reg after = " IF_ARM_ELSE("%d", "%ld") "\n",
+            mc.IF_ARM_ELSE(r10, r28));
+
+    mc.IF_ARM_ELSE(r10, r28) = orig_value;
+
+    dr_set_mcontext(drcontext, &mc);
+}
+
+static void
+save_stolen_reg_to_tls()
```

Comment:
nit suggest: The client shouldn't really be aware of where DR is putting the stolen reg value, whether it's in TLS or somewhere in the private mcontext.  This is a black box test using public interfaces: so maybe s/save_stolen_reg_to_tls/change_stolen_reg_value/ and the other "read_and_restore_stolen_reg_value" or something.

Reply:
Done.